Legacy/bonej/src/main/java/org/bonej/plugins/SliceGeometry.java

```diff
@@ -1032,8 +1061,28 @@ public void run() {
 						}
 					}
 				}
+
 				stdevThick[s] = Math.sqrt(sumSquares / pixCount);
 			}
 		}
 	}
+
+	/**
+	 * Calculate the proportion of a pixel that contains foreground, assuming a
+	 * two-phase image (foreground and background) and linear relationship
+	 * between pixel value and physical density. If the pixel value is greater
+	 * than the foreground value, this method will return 1, and if lower than
+	 * the background value, returns 0.
+	 * 
+	 * @param pixel
+	 *            the input pixel value
+	 * @return fraction of pixel 'size' occupied by foreground
+	 */
+	private double filledFraction(final double pixel) {
+		if (pixel > foreground)
+			return 1;
+		if (pixel < background)
+			return 0;
```

Comment:
if-blocks miss braces

Reply:
Also, it's never checked that `background <= foreground`

Reply:
Done. Input checked during dialog processing.